### PR TITLE
Add platform dependent download link

### PR DESCRIPTION
**Pull request by rubenwardy**
- [ ] confirm user agents:
   - [x] Windows (correct according to MS docs)
   - [x] Ubuntu
   - [x] Linux Mint (according to http://www.webapps-online.com/online-tools/user-agent-strings/dv/operatingsystem301243/linux-mint)
   - [x] Debian (from 1)
   - [x] Fedora (from 1)
   - [ ] Arch
   - [ ] OpenSUSE
   - [ ] Gentoo
   - [ ] Mageia
   - [ ] Void Linux
- [ ] What about out of date versions?
- ~~Make Ubuntu link open in software center~~

1 = https://www.whatismybrowser.com/developers/tools/user-agent-parser/browse/operating-system-name/linux-user-agents

screenshot_2017-03-12_02-11-47

### CODE REVIEW

**Comment by sofar** [comment]
that reminds me we should really do flatpak builds, and advertise those as well.

**Comment by est31** [comment]
@sofar in the case you were not aware, there are already unofficial flatpak builds at the forum: https://forum.minetest.net/viewtopic.php?f=42&t=15336

Github link: https://github.com/est31/minetest-flatpak

What's missing is three things:

1. singing keys (prefferably c55 has a copy so that in the case sth happens we won't lose them)
2. server to host the builds as a flatpak repo. Just having bare flatpak files is not the way flatpak ought to be used. You are supposed to have a repo.
3. (maybe) some automation to have a nightly repo as well

I hoped flathub would have became a thing, but seems nothing is happening here, so we'll have to roll our own. :/

File: downloads.html [hunk 1]
```diff
@@ -16,6 +27,48 @@
...
+        if (platform.indexOf("win") >= 0) {
+            platform_name = "Windows";
+            title = "{{ page.latest_downloads.windows.title }}";
+            url = "{{ page.latest_downloads.windows.url }}";
+        } else if (platform.indexOf("mac") >= 0) {
+            platform_name = "OS X";
+            title = "{{ page.latest_downloads.mac.title }}";
+            url = "{{ page.latest_downloads.mac.url }}";
+        } else if (platform.indexOf("linux") >= 0) {
+            platform_name = "Ubuntu";
```
Comment (sfan5) [inline downloads.html]: **TRIGGERED**

I mean this is pretty stupid, most browsers on Ubuntu even have that in the user agent so you could look for that. Don't even bother trying to add a download button for the other linux flavors

Comment (est31) [inline downloads.html]: I think we should instead redirect to the download page in the case of linux.

Comment (rubenwardy) [inline downloads.html]: It's a good guess, but yeah - it's on my todo list

**Comment by rubenwardy** [comment]
updated

File: downloads.html [hunk 2]
```diff
@@ -16,6 +38,43 @@
...
+
+    <p>
+        <small>scroll down for more versions</small>
+    </p>
+</div>
+
+<script>
+    function detectPlatformName() {
+        var userAgent = navigator.userAgent.toLowerCase();
+        var platforms = ["windows", "mac", "ubuntu", "debian", "fedora"];
```
Comment (sfan5) [inline downloads.html]: is `fedora` or `debian` actually found in user agents?
afaik only ubuntu does this

File: downloads.html [hunk 3]
```diff
@@ -8,6 +8,28 @@
...
+        title: portable, 64-bit
+        url: https://github.com/minetest/minetest/releases/download/0.4.15/minetest-0.4.15-win64.zip
+    mac:
+        os: OS X
+        title: Stable OS X Builds
+        url: https://forum.minetest.net/viewtopic.php?t=9190
+    ubuntu:
+        os: Ubuntu
+        title: search for .deb
+        url: http://packages.ubuntu.com/search?keywords=minetest
```
Comment (sfan5) [inline downloads.html]: might be better to link to the software center:
https://apps.ubuntu.com/cat/applications/minetest/ or a link that would open the "software center" application

File: downloads.html [hunk 4]
```diff
@@ -8,6 +8,28 @@
...
+    debian:
+        os: Debian
+        title: search for .deb
+        url: https://packages.debian.org/search?keywords=minetest
```
Comment (sfan5) [inline downloads.html]: how do we deal with debian & ubuntu mostly having ridiculously outdated versions?

File: downloads.html (hunk 2 above)
Comment (rubenwardy) [inline downloads.html]: according to https://www.whatismybrowser.com/developers/tools/user-agent-parser/browse/operating-system-name/linux-user-agents yes

File: downloads.html (hunk 4 above)
Comment (rubenwardy) [inline downloads.html]: No idea :/
Maybe detect whether the OS is LTS or not for Ubuntu?
Or get the user to install a stable MT apt repo

**Comment by rubenwardy** [comment]
added all Linux platforms in the linux section, although I have no idea if they work (yet) :P

File: downloads.html (hunk 4 above)
Comment (rubenwardy) [inline downloads.html]: It's worth noting that I literally copied the links from the Linux section, and this problem existed before. There's just less text now to warn them

**Comment by rubenwardy** [comment]
Updated.

Worrying about old versions existed before this PR. I think this should be solved in a separate PR by adding apt-repo instructions.

**Comment by Megaf** [comment]
I think we should go the same path big browsers are going.

1. Detect OS family, Linux, Windows, macOS
2. Offer most common package options and architectures for that family. For example. If Linux, offer [Linux 64 DEB], [Linux 64 RPM], [Linux 64 TGZ], [Linux 32 DEB], [Linux 32 RPM], [Linux 32 TGZ], [Source]. If Windows offer, [Windows 64 Installer], [Windows 32 Installer]. If macOS [macOS 64 DMG].

**Comment by benrob0329** [comment]
Should be

```
sudo pacman -S Minetest
```

For arch, as its generally up to date.

**Comment by asddsaz** [comment]
For Elementary OS it could automatically open in the App Store.

**Pull request closed without merging**